Src/LogData.c

```diff
@@ -10,342 +19,241 @@
  */
-void checkEEPROMBlocking(uint16_t timeout)
+void checkEEPROMBlocking()
 {
 	while(HAL_I2C_GetState(&hi2c1) != HAL_I2C_STATE_READY){};
-	while (HAL_I2C_IsDeviceReady(&hi2c1, deviceAddress<<1, 3, timeout) != HAL_OK){};
+	while (HAL_I2C_IsDeviceReady(&hi2c1, DEVICE_ADDRESS<<1, 3, TIMEOUT_MS) != HAL_OK){};
+    // TODO: add a delay ~100us?
+    // should this return a boolean?
 }
 
-void writeAllDataToEEPROM(AllData* data, uint16_t memAddress, uint16_t timeout)
+/**
+ * @brief puts a LogEntry into a char array and sends to EEPROM for writing
+ * @param memAddress, the address at which the EEPROM is instructed to start writing
+ * @param givenLog, pointer to a log initialized at the start of task
+ */
+void writeLogEntryToEEPROM(uint16_t memAddress, LogEntry* givenLog)
 {
-    char dataToWrite[200]; 
-
-    int index = 0;
-    dataToWrite[0] = data->accelGyroMagnetismData_ ->accelX_;
-    dataToWrite[4] = data->accelGyroMagnetismData_ ->accelY_;
-    dataToWrite[8] = data->accelGyroMagnetismData_ ->accelZ_;
-    dataToWrite[12] = data->accelGyroMagnetismData_ ->gyroX_;
-    dataToWrite[16] = data->accelGyroMagnetismData_ ->gyroY_;
-    dataToWrite[20] = data->accelGyroMagnetismData_ ->gyroZ_;
-    dataToWrite[24] = data->accelGyroMagnetismData_ ->magnetoX_;
-    dataToWrite[28] = data->accelGyroMagnetismData_ ->magnetoY_;
-    dataToWrite[32] = data->accelGyroMagnetismData_ ->magnetoZ_;
-
-    dataToWrite[36] = data->barometerData_ ->pressure_;
-    dataToWrite[40] = data->barometerData_ ->temperature_;
-
-    dataToWrite[44] = data->combustionChamberPressureData_ ->pressure_;
-
-    dataToWrite[48] = data->gpsData_ ->time_;
-    dataToWrite[52] = data->gpsData_->latitude_.degrees_;
-    dataToWrite[56] = data->gpsData_->latitude_.minutes_;
-    dataToWrite[60] = data->gpsData_->longitude_.degrees_;
-    dataToWrite[64] = data->gpsData_->longitude_.minutes_;
-
-    dataToWrite[68] = data->gpsData_->antennaAltitude_.altitude_;
-    dataToWrite[72] = data->gpsData_->geoidAltitude_.altitude_;
-    dataToWrite[76] = data->gpsData_->totalAltitude_.altitude_;
-
-
-    dataToWrite[80] = data->oxidizerTankPressureData_ ->pressure_;
-
-    checkEEPROMBlocking(uint16_t timeout);
-    writeToEEPROM(dataToWrite, 80, uint16_t memAddress, uint16_t timeout);
-    // need: memAddress and timeout??????
-    // increment address
+    checkEEPROMBlocking();
+    //Note: writeToEEPROM expects a uint8_t*
+    //if that's not a problem, can we send the LogEntry pointer without casting?
+    writeToEEPROM((char*)givenLog, LOG_ENTRY_SIZE, memAddress);
 }
 
-void readAllDataFromEEPROM(uint16_t addr, AllData* rtn, uint16_t memAddress, uint16_t timeout)
+/**
+ * @brief reads LogEntry-sized char array from EEPROM and updates the log entry
+ * @param memAddress, the address at which the EEPROM is instructed to start reading
+ * @param givenLog, pointer to a log initialized at the start of task
+ */
+void readLogEntryToEEPROM(uint16_t memAddress, LogEntry* givenLog)
```

Comment:
read from ;((((

Reply:
not using dataRead, instead going with this: readFromEEPROM((uint_8*)givenLog, sizeof(givenLog), memAddress); (??? let me know haha)